src/codemagic/tools/app_store_connect.py

```diff
@@ -127,13 +133,20 @@ def _get_resource(self, resource_id, resource_manager, should_print):
         self.printer.print_resource(resource, should_print)
         return resource
 
-    def _list_resources(self, resource_filter, resource_manager, should_print):
+    def _list_resources(self, resource_filter, resource_manager, should_print, application_id=None):
```

Comment:
This method does too much now for my taste. It would be better to separate listing for fixed resource and define new method which would be something like

```python
    def _list_resources_for(self, resource_id: ResourceId, resource_filter, resource_manager, should_print):
        try:
            resources, _included = resource_manager.list(resource_id, resource_filter=resource_filter)
        except AppStoreConnectApiError as api_error:
            raise AppStoreConnectError(str(api_error))
        ...
```

Reply:
as proposed, moved the logic out of `_list_resources` https://github.com/codemagic-ci-cd/cli-tools/pull/67/commits/87fa4db77f534bc9430952896e383ea509531914#diff-c655fbe88a18ddd4d367405c06673842a69855a44563b37bd3b06625bf712419R215